internal/xds/translator/translator_test.go

```diff
@@ -41,357 +41,166 @@ var (
 	overrideTestData = flag.Bool("override-testdata", false, "if override the test output data.")
 )
 
+type testFileConfig struct {
+	dnsDomain                 string
+	requireSecrets            bool
+	requireEnvoyPatchPolicies bool
+	errMsg                    string
+}
+
 func TestTranslateXds(t *testing.T) {
-	testCases := []struct {
-		name                      string
-		dnsDomain                 string
-		requireSecrets            bool
-		requireEnvoyPatchPolicies bool
-		err                       bool
-	}{
-		{
-			name: "empty",
-		},
-		{
-			name: "http-route",
-		},
-		{
-			name: "http-route-regex",
-		},
-		{
-			name: "http-route-redirect",
-		},
-		{
-			name: "http-route-mirror",
-		},
-		{
-			name: "http-route-multiple-mirrors",
-		},
-		{
-			name: "http-route-multiple-matches",
-		},
-		{
-			name: "http-route-direct-response",
-		},
-		{
-			name: "http-route-request-headers",
-		},
-		{
-			name: "http-route-response-add-headers",
-		},
-		{
-			name: "http-route-response-remove-headers",
-		},
-		{
-			name: "http-route-response-add-remove-headers",
-		},
-		{
-			name: "http-route-weighted-invalid-backend",
-		},
-		{
-			name: "http-route-dns-cluster",
-		},
-		{
-			name:           "http-route-with-tls-system-truststore",
+	testConfigs := map[string]testFileConfig{
+		"http-route-with-tls-system-truststore": {
 			requireSecrets: true,
 		},
-		{
-			name:           "http-route-with-tlsbundle",
+		"http-route-with-tlsbundle": {
 			requireSecrets: true,
 		},
-		{
-			name:           "http-route-with-tlsbundle-multiple-certs",
+		"http-route-with-tlsbundle-multiple-certs": {
 			requireSecrets: true,
 		},
-		{
-			name:           "simple-tls",
+		"simple-tls": {
 			requireSecrets: true,
 		},
-		{
-			name:           "mutual-tls",
+		"mutual-tls": {
 			requireSecrets: true,
 		},
-		{
-			name:           "mutual-tls-required-client-certificate-disabled",
+		"mixed-tls-jwt-authn": {
 			requireSecrets: true,
 		},
-		{
-			name:           "http3",
+		"mutual-tls-required-client-certificate-disabled": {
 			requireSecrets: true,
 		},
-		{
-			name: "tls-route-passthrough",
-		},
-		{
-			name: "tcp-route-simple",
-		},
-		{
-			name: "tcp-route-complex",
-		},
-		{
-			name: "tcp-route-tls-terminate",
-		},
-		{
-			name: "multiple-simple-tcp-route-same-port",
-		},
-		{
-			name: "http-route-weighted-backend",
-		},
-		{
-			name: "tcp-route-weighted-backend",
-		},
-		{
-			name:           "multiple-listeners-same-port",
+		"http3": {
 			requireSecrets: true,
 		},
-		{
-			name: "udp-route",
-		},
-		{
-			name: "http2-route",
-		},
-		{
-			name: "http-route-rewrite-url-prefix",
-		},
-		{
-			name: "http-route-rewrite-root-path-url-prefix",
-		},
-		{
-			name: "http-route-rewrite-url-fullpath",
-		},
-		{
-			name: "http-route-rewrite-url-host",
-		},
-		{
-			name: "http-route-timeout",
-		},
-
-		{
-			name: "ratelimit",
+		"multiple-listeners-same-port": {
+			requireSecrets: true,
```

Comment:
can this check also be removed, and secrets be asserted if a secrets file exists ?

Reply:
good idea!

Reply:
by removing `requireSecrets` and checking secrets from translator result, I found some `xxx.secret.yaml` were missing, so added in this PR as well